microbench/ribbon_bench.cc

```diff
@@ -0,0 +1,154 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#include <benchmark/benchmark.h>
+
```

Comment:
This seems like a reasonable starting point, but filter_bench is more representative of how filters are used in RocksDB, e.g. by queries hitting different filters occupying a large memory space. We should express here that the goal is to eventually get those kinds of results into the reusable benchmarking framework. One could get the impression that this file is exclusively focused on "more micro" benchmarking of filter behavior than filter_bench does, but I do not believe that is the goal. Others could assume this file is "authoritative" and no other benchmarking for filters exists.

In broader terms, I fear that even using the term "microbenchmark" might distract from the goal (as I see it) of taking meaningful measurements of subsystem performance to catch regressions (and track improvements) by focusing too much on individual components (e.g. querying a filter) rather than treating them as subsystems (e.g. querying one of many filters).

Reply:
I agree both.
1. this file is just the first benchmark that more for demonstrating the integration. Compare to `filter_bench`, it's missing lots tests, but I guess if we were having this microbenchmark framework before, we might just use it instead of creating stand-alone `filter_bench`. I'll add comment to refer to `filter_bench` for better filter benchmark.
2. For `microbenchmark` focusing too much on small component, yes, I think it should not replace db_bench, it just gives extra, targeted information. it's also up to the test designer, so when writing future benchmark tests, we should have a more meaningful benchmark and metrics. Some metrics might be useful during development (like this one) but maybe not useful (or likely change) after feature is done, which we could exclude in the run or just ignore the result.